rakelib/camelize_file.rake

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+namespace :camelize_file do
+  desc 'Given a json schema file it is transformed into a camelCase version'
+  # example `bundle exec rake camelize_file:schema[user_loa3]`
+  task :schema, [:json_schema_file] => [:environment] do |_, args|
+    raise IOError, 'No json-schema file provided' unless args[:json_schema_file]
+
+    schema_path = Rails.root.join('spec', 'support', 'schemas', "#{args[:json_schema_file]}.json")
+    raise IOError, "No json-schema file at #{schema_path}" unless File.exist? schema_path
+
+    transformer = SchemaCamelizer.new(schema_path.to_s)
+    transformer.save!
+    print "Saved camelized schema to #{transformer.camel_path}\n"
+  end
+end
+
+class SchemaCamelizer
+  attr_reader :original_path
+
+  def initialize(schema_path)
+    @original_path = schema_path
+    raw_schema = File.read(schema_path)
+    raw_schema.gsub!(/"required": \[(["\w*"\,? ?\n?]*)\]/) do
+      # rubocop:disable Style/PerlBackrefs
+      keys = $1.split(',').map(&:strip).map { |key| camelizer.call(key.gsub('"', '')) }
```

Comment:
What is this block doing?

Reply:
i added some comments because there is a lot going on